magefile.go

```diff
@@ -980,8 +1004,28 @@ func packageAgent(platforms []string, dependenciesVersion string, agentPackaging
 		log.Printf("--- Packaging dependenciesVersion[%s], %+v \n", dependenciesVersion, requiredPackages)
 	}
 
+	if devtools.PackagingFromManifest {
+		// If we're using a manifest, download it here, and pass in the manifestResponse
+		// to other functions that need to operate on the manifest
+		if manifestResponse, err = manifest.DownloadManifest(devtools.ManifestURL); err != nil {
+			log.Panicf("failed to download remote manifest file %s", err)
+		} else {
+			if parsedVersion, err := version.ParseVersion(manifestResponse.Version); err != nil {
+				log.Panicf("the manifest version from manifest is not semver, got %s", manifestResponse.Version)
+			} else {
+				// When getting the packageVersion from snapshot we should also update the env of SNAPSHOT=true which is
+				// something that we use as an implicit parameter to various functions
+				if parsedVersion.IsSnapshot() {
+					os.Setenv(snapshotEnv, "true")
+					mage.Snapshot = true
+				}
+				os.Setenv("BEAT_VERSION", parsedVersion.CoreVersion())
+			}
+		}
+	}
```

Comment:
```suggestion
		if manifestResponse, err = manifest.DownloadManifest(devtools.ManifestURL); err != nil {
			log.Panicf("failed to download remote manifest file %s", err)
		}

		if parsedVersion, err := version.ParseVersion(manifestResponse.Version); err != nil {
			log.Panicf("the manifest version from manifest is not semver, got %s", manifestResponse.Version)
		} else {
			// When getting the packageVersion from snapshot we should also update the env of SNAPSHOT=true which is
			// something that we use as an implicit parameter to various functions
			if parsedVersion.IsSnapshot() {
				os.Setenv(snapshotEnv, "true")
				mage.Snapshot = true
			}
			os.Setenv("BEAT_VERSION", parsedVersion.CoreVersion())
		}
	}
```

The else here makes this harder to follow and you should be able to remove it.